doc/member_types.md

```diff
@@ -543,54 +545,50 @@ Patch<SomeObject>():
 ```
 
 
-
 ## Type Modifiers
 
-### `optional`
 
-A member with type modifier `optional` can have the placeholder value
-`None` assigned. `None` indicates that the member is initialized, but
-has no regular value. `None` can only be assigned and not used as a
-second operand in other operations. Additionally, operations other than
-assignment have no effect on members that have `None` assigned.
+### `abstract`
 
-Members of any data type can have the `optional` modifier. `optional`
-members still need to be initialized with either `None` or a regular
-value to not be abstract.
+`abstract` is a type modifier for the `object` data type. It signifies
+that references to abstract nyan objects can be assigned as member values.
 
 ```python
-SomeObject():
-    a : optional(int)                 # declaration of an int as optional
-    b : optional(float)               # declaration of a float as optional
-    c : optional(set(OtherObject))    # declaration of a set(OtherObject) as optional
-    d : optional(OtherObject)         # declaration of a nyan object reference to OtherObject as optional
-
 OtherObject():
-    a : optional(int) = None                 # declaration and initialization with None
-    b : optional(OtherObject) = OtherObject  # declaration and initialization with regular value
+    # is abstract because member 'x' is abstract
+    x : int
+
+ChildObject(OtherObject):
+    # not abstract because member 'x' now has a value
+    x = 5
+
+SomeObject():
+    a : abstract(OtherObject)                # declaration of a to allow abstract objects with type OtherObject
+    b : abstract(OtherObject) = OtherObject  # declaration of b to allow abstract objects with type OtherObject
+                                             # and initialization with ChildObject
```

Comment:
what about assigning `ChildObject` to b, would that work? (yes)

Reply:
example added